internals/plan/extensions_test.go

```diff
@@ -0,0 +1,680 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/plan"
+)
+
+type inputLayer struct {
+	order int
+	label string
+	yaml  string
+}
+
+// PlanResult represents the final content of a combined plan. Since this
+// package exclusively focuses extensions, all built-in sections are empty
+// and ignored in the test results.
+type planResult struct {
+	x *xSection
+	y *ySection
+}
+
+var extensionTests = []struct {
+	extensions             map[string]plan.LayerSectionExtension
+	layers                 []*inputLayer
+	errorString            string
+	combinedPlanResult     *planResult
+	combinedPlanResultYaml string
+}{
+	// Index 0: No Sections
+	{
+		combinedPlanResultYaml: string(reindent(`
+			{}`)),
+	},
+	// Index 1: Sections with empty YAML
+	{
+		extensions: map[string]plan.LayerSectionExtension{
+			"x-field": &xExtension{},
+			"y-field": &yExtension{},
+		},
+		combinedPlanResult: &planResult{
+			x: &xSection{},
+			y: &ySection{},
+		},
+		combinedPlanResultYaml: string(reindent(`
+			{}`)),
+	},
+	// Index 2: Load file layers invalid section
+	{
+		extensions: map[string]plan.LayerSectionExtension{
+			"x-field": &xExtension{},
+			"y-field": &yExtension{},
+		},
+		layers: []*inputLayer{
+			&inputLayer{
```

Comment:
I *think* you can drop the `&inputLayer` on these lines, as the type above already specifies it.

Reply:
Nice. Applied. Resolving.